kroxylicious-filters/kroxylicious-oauthbearer/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.oauthbearer;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.sasl.Sasl;
+
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerValidatorCallbackHandler;
+import org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslServerProvider;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.kroxylicious.proxy.filter.FilterFactory;
+import io.kroxylicious.proxy.filter.FilterFactoryContext;
+import io.kroxylicious.proxy.plugin.Plugin;
+import io.kroxylicious.proxy.plugin.PluginConfigurationException;
+
+@Plugin(configType = OauthBearer.Config.class)
+public class OauthBearer implements FilterFactory<OauthBearer.Config, SharedContext> {
+
+    public static final String OAUTHBEARER_MECHANISM = "OAUTHBEARER";
+
+    @SuppressWarnings("unused")
+    public OauthBearer() {
+        this.oauthHandler = new OAuthBearerValidatorCallbackHandler();
+    }
+
+    public OauthBearer(OAuthBearerValidatorCallbackHandler oauthHandler) {
+        this.oauthHandler = oauthHandler;
+    }
+
+    private final OAuthBearerValidatorCallbackHandler oauthHandler;
+
+    static {
+        OAuthBearerSaslServerProvider.initialize();
+    }
+
+    @Override
+    public SharedContext initialize(FilterFactoryContext context, Config config) throws PluginConfigurationException {
+        try {
+            oauthHandler.configure(
```

Comment:
`OAuthBearerValidatorCallbackHandler` are closeable objects.  Kroxylicious currently has no mechanism to allow the factory to release the resources it has acquired.   This might affect an embedded use-case of Kroxylicious (AFAICS the RefreshingHttpsJwksVerificationKeyResolver would never get shutdown).

This isn't really a comment against this PR, it is really an acknowledgement that FilterFactory ought to have properly lifecycle.

Maybe this PR should add a comment in the code to acknowledge the potential for a leak.

Reply:
@callaertanthony https://github.com/kroxylicious/kroxylicious/commit/b8d8b1aeaccfff215a7c116d043a3c9b0e562631 added support for `FilterFactor.close()` if you want to make use of it.